Library/Homebrew/cmd/tap.rb

```diff
@@ -28,14 +32,16 @@ def install_tap user, repo
     tapd.find_formula { |file| files << file }
     puts "Tapped #{files.length} formula#{plural(files.length, 'e')} (#{tapd.abv})"
 
-    if private_tap?(repouser, repo) then puts <<-EOS.undent
-      It looks like you tapped a private repository. To avoid entering your
-      credentials each time you update, you can use git HTTP credential caching
-      or issue the following command:
+    unless clone_target
```

Comment:
I didn't omit it—or not intentionally at least. 

The `unless clone_target` test checks whether this is the "fancy" style of tapping where we are given an extra argument. If we are given an explicit item to clone, then we are in `any-tap` mode, and so we assume the user knows what they're doing. If we're still in `tap` mode (i.e. the way Homebrew has always done it), then we check for private repos.

tl;dr For people who continue to run only `brew tap`, this should work like it always did.